src/argilla/server/contexts/datasets.py

```diff
@@ -509,83 +519,190 @@ async def _validate_suggestion(
     return questions
 
 
+async def validate_user(db: "AsyncSession", user_id: UUID, users_ids: Optional[Set[UUID]]) -> Set[UUID]:
+    if not users_ids:
+        users_ids = set()
+
+    if user_id not in users_ids:
+        if not await accounts.user_exists(db, user_id):
+            raise ValueError(f"user_id={str(user_id)} does not exist")
+        users_ids.add(user_id)
+
+    return users_ids
+
+
+async def validate_vector(
+    db: "AsyncSession",
+    dataset_id: UUID,
+    vector_name: str,
+    vector_value: List[float],
+    vectors_settings: Optional[Dict[str, VectorSettingsSchema]] = None,
+) -> Dict[str, VectorSettingsSchema]:
+    if vectors_settings is None:
+        vectors_settings = {}
+
+    vector_settings = vectors_settings.get(vector_name, None)
+    if not vector_settings:
+        vector_settings = await get_vector_settings_by_name_and_dataset_id(db, vector_name, dataset_id)
+        if not vector_settings or vector_settings.dataset_id != dataset_id:
```

Comment:
I'm missing something or the second condition will never be `True` ? I mean you are finding the vector_settings by `dataset_id` and `name` so if found it will always belongs to that dataset.

```suggestion
        if not vector_settings:
```

Reply:
Done